src/utils/validateProperties.ts

```diff
@@ -0,0 +1,20 @@
+import { PineconeArgumentError } from '../errors';
+import { CreateCollectionRequest } from '../pinecone-generated-ts-fetch/control';
+
+// Types and consts for property validation of generated Interfaces to ensure no unknown/invalid properties are passed, no req'd properties are missing
```

Comment:
Should this maybe live in the `src/control/createCollection.ts`? Seems like most of the specific constants for validation live in the respective implementation files.

Reply:
Ah yes, great call, let's do it!